clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -514,10 +514,11 @@ private void fetchOffsetsWithRetries(final OffsetFetchRequestState fetchRequest,
 
         // Retry the same fetch request while it fails with RetriableException and the retry timeout hasn't expired.
         currentResult.whenComplete((res, error) -> {
-            boolean inflightRemoved = pendingRequests.inflightOffsetFetches.remove(fetchRequest);
-            if (!inflightRemoved) {
-                log.warn("A duplicated, inflight, request was identified, but unable to find it in the " +
-                    "outbound buffer:" + fetchRequest);
+            if (!fetchRequest.isExpired()) {
+                boolean inflightRemoved = pendingRequests.inflightOffsetFetches.remove(fetchRequest);
+                if (!inflightRemoved) {
+                    log.warn("The response for the offset fetch request for partitions {} was not found in the inflight buffer", fetchRequest.requestedPartitions);
+                }
```

Comment:
When is the request removed from the `inflightOffsetFetches` when it is expired?

Reply:
It is removed later, when the user calls `addOffsetFetchRequest()` with a set of partitions that doesn't match any in the unsent or inflight queues.

However, this change has been reverted, but I'll double check how the new approach handles it.